packages/storybook/stories/utils.ts

```diff
@@ -1,25 +1,135 @@
+import { useEffect, useRef, useState } from 'react';
+
+const renderVideoStream = async (scalingMode?: string): Promise<HTMLElement | null> => {
   if (navigator.mediaDevices?.getUserMedia) {
     const video = document.createElement('video');
     video.autoplay = true;
     video.style.height = '100%';
     video.style.width = '100%';
     video.style.objectFit = scalingMode ?? 'cover';
 
-    navigator.mediaDevices
-      .getUserMedia({ video: true })
-      .then(function (stream) {
-        video.srcObject = stream;
-      })
-      .catch(function (error) {
-        console.log('Something went wrong!', error);
-      });
+    try {
+      video.srcObject = await navigator.mediaDevices.getUserMedia({ video: true });
+    } catch (e) {
+      console.log('Unable to get stream for video', e);
+    }
 
     return video;
   }
 
   return null;
 };
+
+const stopRenderVideoStream = (video: HTMLElement | null): void => {
+  if (!video) {
+    return;
+  }
+
+  const mediaProvider = (video as HTMLVideoElement).srcObject;
+  if (!mediaProvider) {
+    return;
+  }
+
+  const stream = mediaProvider as MediaStream;
+  if (!stream.getVideoTracks) {
+    return;
+  }
+
+  for (const track of stream.getVideoTracks()) {
+    track.stop();
+  }
+};
+
+/**
+ * Manages a single video stream. If enabled is true will create and return a video stream. When component is unmounted
+ * will automatically dispose of the video stream.
+ *
+ * @param enabled - If creating a video stream
+ * @returns
+ */
+export const useVideoStream = (enabled: boolean): HTMLElement | null => {
+  const mounted = useRef(false);
+  useEffect(() => {
+    mounted.current = true;
+    return () => {
+      mounted.current = false;
+    };
+  });
+
+  const [videoStreamElement, setVideoStreamElement] = useState<HTMLElement | null>(null);
+  useEffect(() => {
+    const loadVideo = async (): Promise<void> => {
+      const videoStreamElement = await renderVideoStream();
+      setVideoStreamElement(videoStreamElement);
+      // Since the render is async, it may finish after the component is unmounted already, in this case we have to make
+      // sure to clean up the stream
+      if (!mounted.current) {
+        stopRenderVideoStream(videoStreamElement);
+      }
+    };
+    if (enabled) {
+      loadVideo();
+    }
+  }, [enabled]);
+
+  // Clean up videoStream if the component unmounts
+  useEffect(() => {
+    return () => {
+      stopRenderVideoStream(videoStreamElement);
+    };
+  }, [videoStreamElement]);
+
+  return videoStreamElement;
+};
+
+/**
+ * Manages multiple videoStreams. The amount is based on the given numberOfStreams parameter. When component is
+ * unmounted will automatically dispose of all the managed video streams.
+ *
+ * @param numberOfStreams - Number of video streams to create
+ * @returns
+ */
+export const useVideoStreams = (numberOfStreams: number): (HTMLElement | null)[] => {
```

Comment:
`useVideoStream` and `useVideoStreams` are almost the same.

Could you just implement one using the other?

    export const useVideoStream = (enabled: boolean): (HTMLElement | null) => {
        const streams = useVideoStreams(enabled ? 1 : 0);
        return streams ? streams[0] : null;
    }

Reply:
Sounds good, I've updated the PR to use useVideoStreams only and deleted the other hook.